### PR TITLE
Forms

### PR DESCRIPTION
**Pull request by SaneMyburg**
# Blog app - Add forms
### In this PR: ###
-  I created a method called `current_user` in ApplicationController to make current user data available to all controllers.
    -  It returns the first user from the database.
- I created forms to perform the following functions:
    - Creates a Post on behalf of the `current_user`.
    - Creates a comment on behalf of the `current_user`.
-  I created actions to: 
   - Allow users to add posts.
   - Allow users to add comments to a post.
   - Allow Users to add likes to Posts.

### CODE REVIEW

File: app/controllers/posts_controller.rb [hunk 1]
```diff
@@ -1,11 +1,46 @@
...
+    @post = Post.new
+  end
+
+  def like
+    @post = Post.find(params[:id])
+    @user = @post.author
+    @like = Like.new(author_id: @user.id, post_id: @post.id)
+    @like.save
+    redirect_to user_post_path(user_id: @user.id, id: @post.id)
+  end
```
Comment (cynthiainga) [inline app/controllers/posts_controller.rb]: - [ ] You should have a `like_contoller.rb` file where you should put all the actions for like, the `new` and `create` actions, please kindly fix that and create the appropriate file for like 🙃 😁

File: app/controllers/posts_controller.rb [hunk 2]
```diff
@@ -1,11 +1,46 @@
...
+    @post = Post.new(post_params)
+    @post.author_id = current_user.id
+    @post.comments_counter = 0
+    @post.likes_counter = 0
+    if @post.save
+      redirect_to user_post_path(current_user, @post), notice: 'Post created successfully.'
+    else
+      render :new
+    end
+  end
```
Comment (cynthiainga) [inline app/controllers/posts_controller.rb]: - [ ] I notice that every time I create a post for the second user which is `Lily` on my list the post goes to the first user, I tried to create other users and when I create a post for other users the post goes to the first user `Tom`. Please kindly fix that 🙏🏾 

https://github.com/SaneMyburg/blog-app/assets/74501523/9affaf94-76ac-4e54-a882-e5e023dcb1f2


https://github.com/SaneMyburg/blog-app/assets/74501523/34a316a7-7bbc-49ab-8cce-a438b3c7f608

**Review by cynthiainga — CHANGES_REQUESTED** [review]
Hi @SaneMyburg, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!



## Highlights

- [ ] ApplicationController contains `#current_user` method ✔️ 
- [ ] Correct Gitflow ✔️ 
- [ ] No linters errors ✔️ 

## Required Changes ♻️

- [ ] Please kindly check the comments under the review 👇🏾 👇🏾 

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you consider them as they can improve your code._


Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please remember to tag me in your question so I can get the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/controllers/posts_controller.rb (hunk 2 above)
Comment (SaneMyburg) [reply app/controllers/posts_controller.rb]: @cynthiainga, thanks for the review I appreciate your time taken, however, it is my understanding that the logged-in user, is the first user who is Tom, so Lilly can't create a post.

**Review by Ismailco — APPROVED** [review]
### Hi @SaneMyburg, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

cr-approve


## Highlights

✔️ No linter errors
✔️ Used Gitflow
✔️ Work documented professionally

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**